### PR TITLE
DDPB-3344: Add audit logging for removal of a deputy from a client

### PR DESCRIPTION
**Pull request by acsauk-mojd**
## Purpose
This change adds further fields to the existing `clientDischarged` audit event to give more context and is required to support some future events.

Fixes DDPB-3344

## Approach
After a chat with @aiesoftware, we decided to wrap the Logger in its own service to smooth over testing. Previously when trying to mock out the controller flow the mocking was getting out of control having to go right down the AWS logging stack. This way we only have to mock out the single notice method.

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [x] I have added tests to prove my work
- [ ] The product team have approved these changes

### Frontend
- [ ] I have run an in-browser accessibility test (e.g. WAVE, Lighthouse)
- [ ] There are no deprecated CSS classes noted in the profiler
- [ ] Translations are used and the profiler doesn't identify any missing

### CODE REVIEW

File: client/app/config/services.yml [hunk 1]
```diff
@@ -103,3 +103,11 @@ services:
             $storage: '@AppBundle\Service\File\Storage\S3Storage'
             $siriusApiGatewayClient: '@AppBundle\Service\Client\Sirius\SiriusApiGatewayClient'
             $restClient: '@AppBundle\Service\Client\RestClient'
+
+    DateTime:
+        class: 'DateTime'
```
Comment (acsauk-mojd) [inline client/app/config/services.yml:108]: This was really weird...I couldn't get Symfony to start without defining DateTime as a service to inject into the fake clock class. Does anyone know if there's any way round this?

Comment (aiesoftware) [reply client/app/config/services.yml:108]: You're on the right lines with the date provider so you can test it, but you shouldn't need the `FakeClock`. I've used a similar service in api called `DateTimeProvider` which is a service that wraps `new \DateTime`. I inject that into the whatever needs to generate a date, and to test it I just mock that service and tell it to return a hard coded date in the tests.

Comment (acsauk-mojd) [reply client/app/config/services.yml]: That's much simpler - I've dropped all the clock classes and switched to using the DateTimeProvider instead 👍

**Comment by aiesoftware** [comment]
Great job on the Controller tests

**Pull request merged**